### DISCUSSION TITLE
Z coordinates stay at 0.0 while printing, all layers printed at same level...

### DISCUSSION THREAD

**Discussion by koenvelle** [Jyers/Marlin#272] [Q&A] [answered]
Hi @Jyers , I'm using the prebuilt binary for enderv2 4.2.7 with fast bltouch, 5x5 grid.

I've set the Z-offset to -2.64.  After leveling, I get a first layer that is as flat as my belly was when I was a kid. 

I can control Z movement via the move menu just fine.

However, once the real printing starts, the Z-coordinate stays at zero... i.e. it is printing layer after layer at height 0.0, resulting in, well, a blob.

Any idea what could be causing this ? 

The start of my gcode looks as follows :

;FLAVOR:Marlin
;TIME:918
;Filament used: 0.27361m
;Layer height: 0.2
;MINX:93.75
;MINY:101.25
;MINZ:0.12
;MAXX:141.25
;MAXY:133.75
;MAXZ:30.32
;Generated with Cura_SteamEngine 4.8.0
M140 S50
M105
M190 S50
M104 S205
M105
M109 S205
M82 ;absolute extrusion mode
; Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
M92 E97.8; E-steps setting
G28 
G29
M420 S1 ; restore Mesh

**Comment by Jyers** [comment]
I'm having a hard time imagining what could be the issue here. Is this behavior consistent with all Gcode files you try?

**Comment by koenvelle** [comment]
Hi,

Thanks for your reply. Yes, at least two different files so far.   If I
manually send a command to lift the head during the print, it does it
without problem. Also the off the shelf firmware from creality works fine,
so it is not a hw issue.  Puzzled....

Kind regards

Koen

**Reply by Jyers** [reply]
Is there anything non stock in your setup other than the 4.2.7 board? Also are your running these files off octoprint or the sd card?

**Comment by koenvelle** [comment]
No, it's a plain 4.2.7 board, no other mods beside the bltouch.
I've tried both off octoprint and the sdcard. I'll experiment a bit with
the gcode, with/without leveling, etc, see if it makes a diff....





Op do 18 mrt. 2021 om 23:15 schreef Jyers ***@***.***>:

**Comment by koenvelle** [comment]
So far, nothing I do seems to make a diff. What I do notice is that after
roughly 20 layers, the z axis _does_ start rising.

**Comment by koenvelle** [comment]
The gcode I'm trying to print to test is attached.
I temporarily reverted to creality's 4.2.7 bltouch firmware. I installed
it, only changed to English and that is it.  The gcode homes, levels in a
3x3 grid, uses the built in z-offset (from the gcode), and prints fine out
of the box.
Then reinstalled your 4.2.7 Bltouch 5x5 fast binary.  Only changed the
x-offset of the probe. Print the same g-code.  Homes, levels in 5x5 grid,
starts printing couple of mm above the bed. I use the tune menu in-print to
manually set a z-offset. printing continues on the bed, but for the first
20 layers or so, the z-axis is not raised, and the coordinate on screen
stays 0....



Op vr 19 mrt. 2021 om 09:28 schreef Koen Velle ***@***.***>:

> So far, nothing I do seems to make a diff. What I do notice is that after
> roughly 20 layers, the z axis _does_ start rising.
>

**Reply by Jyers** [reply]
Try a test print without changing the z offset to see if perhaps that could be causing the issue.

**Comment by koenvelle** [comment]
Without z offset in the gcode you mean ?

**Reply by Jyers** [reply]
No, I meant without modifying the offset during the print, but removing them from the gcode could be another thing to test.

**Reply by koenvelle** [reply]
@Jyers, I think I figured it out...  At some point, I tried modifying the Z-offset using M206. That change got saved and seemed to cause the issue.  After a complete reset to default, and instead issuing M851 to set the offset between BLTouch and Nozzle, it seems to work... Pfew.... In  short, don't use M206 to try and set the Z-offset...
@Jyers, thanks for your responses and the great work on the firmware!

**Comment by koenvelle** [comment]
Hi,

Both do not make a difference.

**Comment by xantrk** [comment]
I just had a similar problem, using vase mode my z axis stopped rising after a couple layers. at z=2.42, manual probing ubl, z offset -0.02. 

EDIT: My issue fixed with disabling arcwelder. It appears that it could be an issue with 3-d arcs.

**Comment by koenvelle** [comment, marked as answer]
, I think I figured it out... At some point, I tried modifying the Z-offset using M206. That change got saved and seemed to cause the issue. After a complete reset to default, and instead issuing M851 to set the offset between BLTouch and Nozzle, it seems to work... Pfew.... In short, don't use M206 to try and set the Z-offset...
@Jyers, thanks for your responses and the great work on the firmware!

**Reply by Jyers** [reply]
Glad you found the issue! Happy printing :)
